ibis_substrait/compiler/translate.py

```diff
@@ -857,6 +845,74 @@ def join(
         )
     )
 
+@translate.register(ops.AsOfJoin)
+def asof_join(
+    op: ops.AsOfJoin,
+    expr: ir.TableExpr,
+    compiler: SubstraitCompiler,
+    **kwargs: Any,
+) -> stalg.Rel:
+    # TODO: Throw on multiple predicates for now (until supported)
+    pred = op.predicates[0]
+
+    left = pred.op().left
+    right = pred.op().right
+
+    left_tr = extrel.AsOfJoinRel.AsOfJoinKeys(on=translate(left.op(), expr, compiler))
+    right_tr = extrel.AsOfJoinRel.AsOfJoinKeys(on=translate(right.op(), expr, compiler))
+
+    detail = pb2_any()
+    detail.Pack(extrel.AsOfJoinRel(
+                    input_keys=[left_tr, right_tr],
+                    tolerance=100
+                ),
+                "" # HACK: Clear out prefix from @type
+    )
+
+    return stalg.Rel(
+        extension_multi=stalg.ExtensionMultiRel(
+            inputs=[
+                translate(op.left, compiler, **kwargs),
+                translate(op.right, compiler, **kwargs)
+            ],
+            detail=detail
+        )
+    )
```

Comment:
This seems generally fine -- we'll have to think about how we want to handle translation rules that are linked to non-first-party extensions.

Reply:
@icexelloss Can this live in your library that is using ibis instead of here? This particular implementation is highly specific to a particular choice of representation at the moment.

Reply:
This function was implemented by Bryce and I as a way to test whether the asof join extension relation could be used successfully. We don't have to include this.

Reply:
I don't have to include the translation rule but I think it's useful for reproducing issues. For example, issues like https://github.com/ibis-project/ibis-substrait/issues/423 can benefit from such rule so that we can at least run asof join with ibis and Acero in the open source